pkg/storage/stores/shipper/indexgateway/gateway.go

```diff
@@ -1,30 +1,34 @@
 package indexgateway
 
 import (
-	"sync"
-
+	"context"
 	"github.com/grafana/dskit/services"
 
 	"github.com/grafana/loki/pkg/storage/chunk"
-	"github.com/grafana/loki/pkg/storage/stores/shipper"
+	chunk_util "github.com/grafana/loki/pkg/storage/chunk/util"
 	"github.com/grafana/loki/pkg/storage/stores/shipper/indexgateway/indexgatewaypb"
 	"github.com/grafana/loki/pkg/storage/stores/shipper/util"
 )
 
 const maxIndexEntriesPerResponse = 1000
 
+type IndexQuerier interface {
+	QueryPages(ctx context.Context, queries []chunk.IndexQuery, callback chunk_util.Callback) error
+	Stop()
+}
+
 type gateway struct {
 	services.Service
 
-	shipper chunk.IndexClient
+	indexQuerier IndexQuerier
 }
 
-func NewIndexGateway(shipperIndexClient *shipper.Shipper) *gateway {
+func NewIndexGateway(indexQuerier IndexQuerier) *gateway {
```

Comment:
suggestion:  maybe you can use `chunk.IndexClient` here? It has the following interface:
```go
type IndexClient interface {
    Stop()

    // For the write path.
    NewWriteBatch() WriteBatch
    BatchWrite(context.Context, WriteBatch) error

    // For the read path.
    QueryPages(ctx context.Context, queries []IndexQuery, callback func(IndexQuery, ReadBatch) (shouldContinue bool)) error
}
```

Reply:
I have dropped the changes from this file after talking to Cyril. I will do this refactoring in a different PR. Thanks for the review!